src/search/services/unified_search_service.py

```diff
@@ -369,6 +265,65 @@ def _apply_sort(self, search: Search, sort: str) -> Search:
 
         return search
 
+    def _log_first_hit_index(self, response) -> None:
+        """Log the index of the first hit if results are available."""
+        if response.hits.total.value > 0:
+            first_hit_index = response.hits[0].meta.index if response.hits else "N/A"
+            logger.info(f"First hit index: {first_hit_index}")
+
+    def _extract_opensearch_error_details(self, exception: Exception) -> dict[str, Any]:
```

Comment:
Nit: Can we move these error functions to a separate util so that they don't add bloat to the builder?